tests/e2e/scale/test_scale_pvc_pod_multi_project.py

```diff
@@ -0,0 +1,145 @@
+
+from tests import helpers
+from ocs_ci.ocs import constants, cluster, machine, node
+from ocs_ci.ocs.resources import pod, pvc
+from ocs_ci.ocs.exceptions import UnexpectedBehaviour, CephHealthException
+from ocs_ci.framework.testlib import scale, E2ETest, ignore_leftovers
+
+log = logging.getLogger(__name__)
+
+
+@scale
+@ignore_leftovers
+@pytest.mark.parametrize(
+    argnames="start_io",
+    argvalues=[
+        pytest.param(
+            *[False], marks=pytest.mark.polarion_id("OCS-1357")
+        ),
+        pytest.param(
+            *[True], marks=pytest.mark.polarion_id("OCS-1357")
+        )
+    ]
+)
+class TestMultiProjectScalePVCPOD(E2ETest):
+    """
+    Scale the OCS cluster to reach 1500 PVC+POD
+    """
+    @pytest.fixture()
+    def setup_fixture(self, request):
+        def finalizer():
+            self.cleanup()
+
+        request.addfinalizer(finalizer)
+
+    def test_multi_project_scale_pvcs_pods(self, setup_fixture, start_io):
+        """
+        Test case to scale PVC+POD with multi projects and reach expected PVC count
+        :param setup_fixture:
+        :param start_io:
+        :return:
+        """
+
+        # Set PVC count based on number of OCS workers
+        ocs_nodes = node.get_typed_nodes(node_type='worker')
+        pvc_count = {3: 1500, 6: 3000, 9: 4500}
+        scale_count = pvc_count[len(ocs_nodes)]
+
+        # Other variables
+        rbd_sc_obj = helpers.default_storage_class(constants.CEPHBLOCKPOOL)
+        cephfs_sc_obj = helpers.default_storage_class(constants.CEPHFILESYSTEM)
+        pvc_count_each_itr = 5
+        size = f"{random.randrange(5, 105, 5)}Gi"
+        fio_size_param = '2G'
+        fio_rate_param = '16k'
+        self.namespace_list, all_pod_obj = ([] for i in range(2))
+
+        # Pre-requisite check for number of OSD's and app worker nodes.
+        helpers.add_required_osd_count(total_osd_nos=3)
+
+        # Create machineset for app worker nodes, which will create one app worker node
+        self.ms_name = machine.create_custom_machineset(instance_type='m5.4xlarge', zone='a')
+        machine.wait_for_new_node_to_be_ready(self.ms_name)
+        self.app_worker_nodes = machine.get_machine_from_machineset(self.ms_name)
+
+        # Create namespace
+        self.namespace_list.append(helpers.create_project())
+
+        # Continue to iterate till the scale pvc limit is reached
+        while True:
+            if scale_count <= len(all_pod_obj):
+                log.info(f"Scaled {scale_count} pvc and pods")
+                break
+            else:
+                log.info(f"Create {pvc_count_each_itr} pods & pvc")
+                pod_obj, pvc_obj = helpers.create_multi_pvc_pod(
+                    self.namespace_list[-1].name, rbd_sc_obj, cephfs_sc_obj, pvc_count_each_itr,
+                    size, fio_rate=fio_rate_param, start_io=start_io, fio_size=fio_size_param,
+                    fio_runtime=3600, node_selector=constants.SCALE_NODE_SELECTOR
```

Comment:
isn't it too much to run 1 hour?

Reply:
Expecting test case to run for more than 3hrs, since the config is going to create 1500/3000/4500 PVC+POD sequentially, but I should reduce the value for IO's at the end of test.